added descriptions of CLI args for configuring Execution Engine options (#967)
hyperledger/besu-docs: docs/Reference/CLI/CLI-Syntax.md

    You can override the default DNS server if it's unreliable or doesn't serve TCP DNS requests, using the
    [experimental option](#xhelp) `--Xp2p-dns-discovery-server=<HOST>`.

### `engine-rpc-http-port`

=== "Syntax"

    ```bash
    --engine-rpc-http-port=<PORT>
    ```

=== "Example"

    ```bash
    --engine-rpc-http-port=8550
    ```

=== "Environment variable"

    ```bash
    BESU_ENGINE_RPC_HTTP_PORT=8550
    ```

=== "Configuration file"

    ```bash
    engine-rpc-http-port=8550
    ```

> The listening port for the Engine API calls (ENGINE, ETH) for JSON-RPC over HTTP.

Reviewer: Is there a default port?

Reviewer 2: default is 8550

> The listening port for the Engine API calls (ENGINE, ETH) for JSON-RPC over HTTP.

Reviewer 3: ```suggestion
The listening port for the Engine API calls (`ENGINE`, `ETH`) for JSON-RPC over HTTP.
```

> The listening port for the Engine API calls (`ENGINE`, `ETH`) for JSON-RPC over HTTP.

Reviewer 4: @jflo, Are Engine API calls, calls made from a consensus client (Teku)? Are these a seperate set of APIs that are implemented for the merge?

### `engine-rpc-ws-port`

=== "Syntax"

    ```bash
    --engine-rpc-ws-port=<PORT>
    ```

=== "Example"

    ```bash
    --engine-rpc-ws-port=8551
    ```

=== "Environment variable"

    ```bash
    BESU_ENGINE_RPC_WS_PORT=8551
    ```

=== "Configuration file"

    ```bash
    engine-rpc-ws-port=8551
    ```

> The listening port for the Engine API calls (ENGINE, ETH) for JSON-RPC over WebSockets.

Reviewer: Same, is there a default port?

Reviewer 2: 8551

> The listening port for the Engine API calls (ENGINE, ETH) for JSON-RPC over WebSockets.

Reviewer 3: ```suggestion
The listening port for the Engine API calls (`ENGINE`, `ETH`) for JSON-RPC over WebSockets.
```

### `engine-host-allowlist`

=== "Syntax"

    ```bash
    --engine-host-allowlist=<hostname>[,<hostname>...]... or "*"
    ```

=== "Example"

    ```bash
    --engine-host-allowlist=localhost,127.0.0.1
    ```

=== "Environment variable"

    ```bash
    BESU_ENGINE_HOST_ALLOWLIST=localhost,127.0.0.1
    ```

=== "Configuration file"

    ```bash
    engine-host-allowlist=localhost,127.0.0.1
    ```

> A comma separated list of hostnames to allow for ENGINE API access (applies to both HTTP and WebSockets).

Reviewer 3: ```suggestion
A comma-separated list of hostnames to allow for `ENGINE` API access (applies to both HTTP and WebSockets).
```

!!!tip

    To allow all hostnames, use `"*"`. We don't recommend allowing all hostnames for production
    environments.

### `engine-jwt-enabled`

=== "Syntax"

    ```bash
    --engine-jwt-enabled
    ```

=== "Example"

    ```bash
    --engine-jwt-enabled=true
    ```

=== "Environment variable"

    ```bash
    BESU_ENGINE_JWT_ENABLED=true
    ```

=== "Configuration file"

    ```bash
    engine-jwt-enabled=true
    ```

> Enables or disables authentication for Engine APIs. Defaults to `false`.

Reviewer 3: ```suggestion
Enables or disables authentication for Engine APIs. The default is `false`.
```
